### PR TITLE
Add one common docker package

### PR DESCRIPTION
**Pull request by zguesmi**

### CODE REVIEW

File: src/main/java/com/iexec/common/docker/client/DockerClientFactory.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.common.docker.client;
+
+public abstract class DockerClientFactory {
+
+    private static DockerClientInstance defaultClient;
+    private static DockerClientInstance authenticatedClient;
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/docker/client/DockerClientFactory.java]: A map is probably more appropriate for auth clients

**Review by james-toussaint — APPROVED** [review]

File: src/main/java/com/iexec/common/docker/client/DockerClientFactory.java (hunk 1 above)
Comment (zguesmi) [reply src/main/java/com/iexec/common/docker/client/DockerClientFactory.java]: My bad, ugly copy-paste. Sorry!

File: src/main/java/com/iexec/common/docker/client/DockerClientInstance.java [hunk 2]
```diff
@@ -0,0 +1,663 @@
...
+            return false;
+        }
+    }
+
+    /**
+     * Docker container
+     * 
+     */
+
+    public DockerRunResponse run(DockerRunRequest dockerRunRequest) {
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/docker/client/DockerClientInstance.java:313]: Will be hard to test

Comment (zguesmi) [reply src/main/java/com/iexec/common/docker/client/DockerClientInstance.java:313]: I agree but we don't have a choice for now.
PS: tests coming in the next PR.

**Review by james-toussaint — APPROVED** [review]

**Pull request merged**